services/security.ts

```
@@ -15,3 +15,16 @@ export const isValidUrl = (url: string): boolean => {
+
+/**
+ * Sanitizes input by removing non-printable control characters.
+ * Preserves tabs (\t), newlines (\n), and carriage returns (\r).
+ * This helps prevent log forging and some injection attacks.
+ */
+export const sanitizeInput = (input: string): string => {
+  if (!input) return '';
+  // Remove non-printable control characters (ASCII 0-31), except for:
+  // \t (9), \n (10), \r (13)
+  // Also remove delete (127)
+  return input.replace(/[\x00-\x08\x0B\x0C\x0E-\x1F\x7F]/g, '');
```

Biome (2.3.13)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)